lib/dockerregistry/types.go

```diff
@@ -185,6 +185,13 @@ type Manifest struct {
 	// the promotion itself is performed, do we "denormalize" at the very last
 	// moment by modifying the argument to each destination path.
 	Renames []Rename `yaml:"renames,omitempty"`
+
+	// Hidden fields; these are data structure optimizations that are populated
+	// from the fields above. As they are redundant, there is no point in
+	// storing this information in YAML.
+	RenamesDenormalized RenamesDenormalized `yaml:"-"`
+	SrcRegistry         *RegistryContext    `yaml:"-"`
+	Filepath            string              `yaml:"-"`
```

Comment:
Cool.  You could also make these private & have methods, which would then let you hide the fact that they are cached.  But this is a nice change.

Reply:
Ah. Yes that would be a lot cleaner. I can't think of a good reason why we would want these to be public vs just private as you suggested. Plus the yaml-hiding thing felt ugly and hacky on its own anyway. Ack.